source/Calamari.Aws/AwsCommand.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Amazon.IdentityManagement;
+using Amazon.IdentityManagement.Model;
+using Amazon.Runtime;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+using Calamari.Commands.Support;
+using Calamari.Deployment;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Aws
+{
+    public abstract class AwsCommand : ICommand
+    {
+        static readonly Regex ArnNameRe = new Regex("^.*?/(.+)$");
+
+        protected readonly ILog log;
+        protected readonly IVariables variables;
+
+        readonly IAmazonSecurityTokenService amazonSecurityTokenService;
+        readonly IAmazonIdentityManagementService amazonIdentityManagementService;
+
+        protected AwsCommand(
+            ILog log,
+            IVariables variables,
+            IAmazonSecurityTokenService amazonSecurityTokenService,
+            IAmazonIdentityManagementService amazonIdentityManagementService)
+        {
+            this.log = log;
+            this.variables = variables;
+            this.amazonSecurityTokenService = amazonSecurityTokenService;
+            this.amazonIdentityManagementService = amazonIdentityManagementService;
+        }
+
+        public int Execute()
+        {
+            LogAwsUserInfo().ConfigureAwait(false).GetAwaiter().GetResult();
+
+            var pathToPackage = new PathToPackage(Path.GetFullPath(variables.Get("package")));
+
+            Execute(new RunningDeployment(pathToPackage, variables));
+
+            return 0;
+        }
+
+        protected abstract void Execute(RunningDeployment deployment);
+
+        async Task LogAwsUserInfo()
+        {
+            if (variables.IsSet(SpecialVariables.Action.Aws.AssumeRoleARN) ||
```

Comment:
It's hard to understand when this would be true. Perhaps assign to separate variables before the `if` statement